linkedList.js

```diff
@@ -0,0 +1,372 @@
+class Node {
+  constructor(data) {
+    this._data = data;
+    this.nextNode = null;
+  }
+
+  getData() {
+    return this._data;
+  }
+
+
+}
+
+class LinkedList {
+  constructor() {
+    this.head = null;
+  }
+
+  length() {
+    var currentNode = this.head;
+    var length = 0;
+
+    if(!currentNode) {
+      return null;
+    }
+
+    while(currentNode) {
+      length++;
+      currentNode = currentNode.nextNode;
+    }
+    return length;
+  }
+
+  insert(value) {
+    var node = new Node(value);
+    var currentNode = this.head;
+
+    if(!currentNode) {
+      this.head = node;
+
+      return node;
+    }
+
+    currentNode = this.head;
+    node.nextNode = currentNode;
+    this.head = node;
+
+    this._length++;
+    return node;
+  }
+
+  search(value) {
+    var currentNode = this.head;
+    if(!currentNode) {
+      return false;
+    }
+
+    while(currentNode.nextNode) {
```

Comment:
Added optimization: if you change the condition on line 58 to `while (currentNode) {`, then you could delete lines 54 through 56 and the method will still work as expected.